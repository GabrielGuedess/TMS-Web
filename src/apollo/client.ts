import { getServerSession } from 'next-auth';

import { cache } from 'react';

import { setContext } from '@apollo/client/link/context';
import { from, type ApolloClient } from '@apollo/client';
import createUploadLink from 'apollo-upload-client/createUploadLink.mjs';
import {
  NextSSRApolloClient,
  NextSSRInMemoryCache,
} from '@apollo/experimental-nextjs-app-support/ssr';

import { authOptions } from 'lib/auth';

const httpLink = createUploadLink({
  uri: process.env.NEXT_PUBLIC_API_URL,
});

function registerApolloClient(client: () => ApolloClient<unknown>) {
  const apollo = cache(client);

  return {
    apollo,
  };
}

const authLink = setContext(async (_, { headers }) => {
  const session = await getServerSession(authOptions);

  return {
    headers: {
      ...headers,
      authorization: session ? `Bearer ${session.token}` : '',
    } as Headers,
  };
});

export const { apollo } = registerApolloClient(
  () =>
    new NextSSRApolloClient({
      link: from([authLink, httpLink]),
      cache: new NextSSRInMemoryCache(),
    }),
);
