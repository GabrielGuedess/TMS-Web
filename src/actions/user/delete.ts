'use server';

import { redirect } from 'next/navigation';

import { apollo } from 'apollo/client';

import { MUTATION_DELETE_USER } from 'graphql/mutations/user/delete';
import {
  type DeleteUserMutation,
  type DeleteUserMutationVariables,
} from 'graphql/generated';

import * as Sentry from '@sentry/nextjs';

export const deleteUser = async (variables: DeleteUserMutationVariables) => {
  await Sentry.withServerActionInstrumentation('deleteUser', async () => {
    await apollo().mutate<DeleteUserMutation, DeleteUserMutationVariables>({
      variables,
      mutation: MUTATION_DELETE_USER,
    });

    redirect('/dashboard/collaborators');
  });
};
