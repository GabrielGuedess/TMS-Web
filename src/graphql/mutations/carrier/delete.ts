import { gql } from '@apollo/client';

export const MUTATION_DELETE_CARRIER_COMPANY = gql`
  mutation DeleteCarrierCompany($deleteCarrierCompanyId: String!) {
    deleteCarrierCompany(id: $deleteCarrierCompanyId) {
      created_at
      created_by
      id
      legalPersonId
      rntrc
      updated_at
      updated_by
    }
  }
`;
