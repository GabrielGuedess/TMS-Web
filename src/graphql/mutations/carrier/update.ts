import { gql } from '@apollo/client';

export const MUTATION_UPDATE_CARRIER_COMPANY = gql`
  mutation UpdateCarrierCompany(
    $id: String!
    $data: CarrierCompanyUpdateInput!
  ) {
    updateCarriercompany(data: $data, id: $id) {
      id
      legalPersonId
      rntrc
      created_at
      created_by
      updated_at
      updated_by
    }
  }
`;
