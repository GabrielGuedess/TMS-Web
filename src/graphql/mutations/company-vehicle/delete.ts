import { gql } from '@apollo/client';

export const MUTATION_DELETE_COMPANY_VEHICLE = gql`
  mutation DeleteCompanyVehicle($deleteCompanyVehicleId: String!) {
    deleteCompanyVehicle(id: $deleteCompanyVehicleId) {
      carrier_company_id
      created_at
      created_by
      id
      updated_at
      updated_by
      vehicle_id
    }
  }
`;
