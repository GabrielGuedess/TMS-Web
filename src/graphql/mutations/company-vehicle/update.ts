import { gql } from '@apollo/client';

export const MUTATION_UPDATE_COMPANY_VEHICLE = gql`
  mutation UpdatedCompanyVehicle(
    $updatedCompanyVehicleId: String!
    $outsourced: CompanyVehicleUpdateInput!
  ) {
    updatedCompanyVehicle(
      id: $updatedCompanyVehicleId
      outsourced: $outsourced
    ) {
      carrier_company_id
      created_at
      created_by
      id
      updated_at
      updated_by
      vehicle_id
    }
  }
`;
