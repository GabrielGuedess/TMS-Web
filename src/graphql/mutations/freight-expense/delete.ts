import { gql } from '@apollo/client';

export const MUTATION_DELETE_FREIGHT_EXPENSE = gql`
  mutation DeleteFreightExpense($delData: DeletFreightExpenseInput!) {
    deleteFreightExpense(delData: $delData) {
      expenseName
      id
      legalClientOrderId
      physicalCustomerOrderId
      value
    }
  }
`;
