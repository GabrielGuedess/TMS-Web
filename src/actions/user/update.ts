'use server';

import { apollo } from 'apollo/client';

import { MUTATION_UPDATE_USER } from 'graphql/mutations/user/update';
import {
  type UpdateUserMutation,
  type UpdateUserMutationVariables,
} from 'graphql/generated';

import * as Sentry from '@sentry/nextjs';

export const updateUser = async (variables: UpdateUserMutationVariables) => {
  const result = await Sentry.withServerActionInstrumentation(
    'updateUser',
    async () => {
      const data = await apollo().mutate<
        UpdateUserMutation,
        UpdateUserMutationVariables
      >({
        variables,
        mutation: MUTATION_UPDATE_USER,
      });

      return data;
    },
  );

  return result;
};
