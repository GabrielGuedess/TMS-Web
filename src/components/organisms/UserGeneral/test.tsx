import { type UserQuery } from 'graphql/generated';

import { toast } from 'sonner';
import { ApolloError } from '@apollo/client';
import { updateUser } from 'actions/user/update';
import { deleteUser } from 'actions/user/delete';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

import { UserGeneral } from '.';

const createObjectURL = jest.fn();

global.URL.createObjectURL = createObjectURL;

jest.mock('sonner', () => ({
  toast: { error: jest.fn() },
}));

jest.mock('actions/user/update', () => ({
  updateUser: jest.fn(),
}));

jest.mock('actions/user/delete', () => ({
  deleteUser: jest.fn(),
}));

const defaultData: UserQuery = {
  user: {
    id: '1',
    name: 'test',
    role: 'ADMIN',
    password: '123',
    username: 'test',
    email: 'test@gmail.com',
    updated_at: new Date(),
    created_at: new Date(),
  },
};

describe('<UserGeneral />', () => {
  it('should render the collaborator general correctly', () => {
    const { container } = render(<UserGeneral data={defaultData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the collaborator general', () => {
    const { container } = render(<UserGeneral data={defaultData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able to update the collaborator', async () => {
    render(<UserGeneral data={defaultData} />);

    const inputDocument = screen.getByLabelText('Document');
    const inputTry = screen.getByLabelText('Try Count');
    const buttonUpdate = screen.getByLabelText('Update User');

    await userEvent.clear(inputDocument);
    await userEvent.clear(inputTry);

    await userEvent.type(inputDocument, '123456789');
    await userEvent.type(inputTry, '3');
    await userEvent.click(buttonUpdate);

    expect(inputDocument).toHaveValue('123456789');
    expect(inputTry).toHaveValue(3);

    expect(updateUser).toHaveBeenCalled();
  });

  it('should be able to delete the collaborator', async () => {
    render(<UserGeneral data={defaultData} />);

    const buttonDeleteTrigger = screen.getByLabelText('Delete Trigger');

    await userEvent.click(buttonDeleteTrigger);

    const buttonDelete = screen.getByLabelText('Delete User');

    await userEvent.click(buttonDelete);

    expect(deleteUser).toHaveBeenCalled();
  });

  it('should be able to click upload avatar in the collaborator', async () => {
    render(<UserGeneral data={defaultData} />);

    const input = screen.getByLabelText('File');
    const buttonUpload = screen.getByLabelText('Change Photo');

    const file = new File([new ArrayBuffer(1)], 'file.png', {
      type: 'image/png',
    });

    Object.defineProperty(file, 'size', { value: 1e6 });

    await userEvent.click(buttonUpload);
    await userEvent.upload(input, file);

    expect(createObjectURL).toHaveBeenCalled();
  });

  it('should be able to upload avatar in the collaborator', async () => {
    render(<UserGeneral data={defaultData} />);

    const input = screen.getByLabelText('File');

    const file = new File([new ArrayBuffer(1)], 'file.png', {
      type: 'image/png',
    });

    await userEvent.upload(input, file);

    expect(createObjectURL).toHaveBeenCalled();
  });

  it('should be able to view uploaded avatar in the collaborator', async () => {
    render(<UserGeneral data={defaultData} />);

    const input = screen.getByLabelText('File');

    const file = new File([new ArrayBuffer(7)], 'test.png', {
      type: 'image/png',
    });

    await userEvent.upload(input, file);
    expect(createObjectURL).toHaveBeenCalled();

    const image = screen.getByLabelText('New Photo');

    expect(image).toBeInTheDocument();
  });

  it('should not be able to upload avatar when file is larger in the collaborator', async () => {
    render(<UserGeneral data={defaultData} />);

    const input = screen.getByLabelText('File');

    const file = new File([new ArrayBuffer(1)], 'file.jpg', {
      type: 'image/png',
    });

    Object.defineProperty(file, 'size', { value: 8e6 });

    await userEvent.upload(input, file);

    expect(toast.error).toHaveBeenCalled();
  });

  it('should not be able to update the collaborator with error', async () => {
    (updateUser as jest.Mock).mockRejectedValueOnce(
      new ApolloError({ errorMessage: 'Server Down' }),
    );

    render(<UserGeneral data={defaultData} />);

    await expect(async () => {
      await updateUser({
        updateUserId: '1',
        userUpdate: { email: 'test2@gamil.com' },
      });

      expect(toast.error).toHaveBeenCalled();
    }).rejects.toThrow();
  });
});
