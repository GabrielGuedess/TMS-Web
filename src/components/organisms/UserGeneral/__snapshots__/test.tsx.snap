// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollaboratorGeneral /> should render the collaborator general 1`] = `
<div
  class="flex flex-col gap-6 md:grid md:grid-cols-12"
>
  <div
    class="col-span-4 flex flex-col items-center justify-center gap-6 rounded-2xl border border-zumthor-100 bg-white-lilac-50 px-6 pb-10 pt-20 shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950"
  >
    <div
      class="relative flex size-36 items-center justify-center rounded-full border border-dashed border-dark-200 dark:border-shark-700"
    >
      <svg
        aria-label="Default Photo"
        class="text-comet-500 transition-all dark:text-dark-500"
        fill="none"
        height="56"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="56"
      >
        <path
          d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="12"
          cy="7"
          r="4"
        />
      </svg>
      <input
        accept="image/*"
        aria-hidden="true"
        aria-label="File"
        class="hidden"
        type="file"
      />
      <button
        aria-label="Change Photo"
        class="group absolute inset-2 z-20 flex cursor-pointer items-center justify-center rounded-full text-white-lilac-50 transition-all hover:bg-black/70 focus:bg-black/70"
        type="button"
      >
        <svg
          class="opacity-0 transition-all group-hover:opacity-100 group-focus:opacity-100"
          fill="none"
          height="36"
          viewBox="0 0 24 24"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.75992 22H17.2399C19.9999 22 21.0999 20.31 21.2299 18.25L21.7499 9.99C21.8899 7.83 20.1699 6 17.9999 6C17.3899 6 16.8299 5.65 16.5499 5.11L15.8299 3.66C15.3699 2.75 14.1699 2 13.1499 2H10.8599C9.82992 2 8.62992 2.75 8.16992 3.66L7.44992 5.11C7.16992 5.65 6.60992 6 5.99992 6C3.82992 6 2.10992 7.83 2.24992 9.99L2.76992 18.25C2.88992 20.31 3.99992 22 6.75992 22Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M10.5 8H13.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M12 18C13.79 18 15.25 16.54 15.25 14.75C15.25 12.96 13.79 11.5 12 11.5C10.21 11.5 8.75 12.96 8.75 14.75C8.75 16.54 10.21 18 12 18Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </button>
    </div>
    <span
      class="text-center text-xs text-comet-500 transition-all dark:text-dark-300"
    >
      Permitido *.jpeg, *.jpg, *.png, *.gif 
      <br />
      tamanho máximo de 5 Mb
    </span>
    <div
      class="flex gap-4"
    >
      <label
        class="relative flex cursor-pointer items-center gap-2"
        for="active"
      >
        <span
          class="text-sm text-comet-500 dark:text-gray-300"
        >
          Ativo
        </span>
        <button
          aria-checked="true"
          aria-label="Active"
          class="peer relative h-5 w-9 rounded-full bg-porcelain-100 transition-all peer-checked:after:translate-x-full peer-focus:outline-none data-[state='checked']:bg-primary-400 dark:border-gray-600 dark:bg-shark-950 data-[state='checked']:dark:bg-primary-400"
          data-state="checked"
          id="active"
          role="switch"
          type="button"
          value="on"
        >
          <span
            class="mx-[2px] flex size-4 rounded-full bg-white-lilac-50 transition-all data-[state='checked']:translate-x-full"
            data-state="checked"
          />
        </button>
      </label>
      <label
        class="relative flex cursor-pointer items-center gap-2"
        for="validated"
      >
        <span
          class="text-sm text-comet-500 dark:text-gray-300"
        >
          Validado
        </span>
        <button
          aria-checked="true"
          aria-label="Validated"
          class="peer relative h-5 w-9 rounded-full bg-porcelain-100 transition-all peer-checked:after:translate-x-full peer-focus:outline-none data-[state='checked']:bg-primary-400 dark:border-gray-600 dark:bg-shark-950 data-[state='checked']:dark:bg-primary-400"
          data-state="checked"
          id="validated"
          role="switch"
          type="button"
          value="on"
        >
          <span
            class="mx-[2px] flex size-4 rounded-full bg-white-lilac-50 transition-all data-[state='checked']:translate-x-full"
            data-state="checked"
          />
        </button>
      </label>
    </div>
    <button
      aria-controls="radix-:r7:"
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="Delete Trigger"
      class="relative transition-transform duration-200 ease-in-out hover:scale-105 focus:scale-105 active:scale-90 min-w-[9.18rem]"
      data-state="closed"
      type="button"
    >
      <div
        aria-label="Button Background"
        class="absolute size-full rounded-md opacity-[16%] bg-danger-500"
      />
      <div
        class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-danger-500"
      >
        <span
          class="z-10 font-medium leading-[1.125rem] tracking-[0.027rem] text-sm"
        >
          Deletar Usuário
        </span>
      </div>
    </button>
  </div>
  <form
    aria-label="Form"
    class="col-span-8 flex flex-col justify-between gap-6 rounded-2xl border border-zumthor-100 bg-white-lilac-50 p-6 shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950"
  >
    <div
      class="grid grid-cols-auto-fill gap-x-4 gap-y-10"
    >
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            aria-label="Document"
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":ra:"
            name="document"
            required=""
            value="123456789"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":ra:"
          >
            Documento
          </label>
        </div>
      </div>
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":rb:"
            readonly=""
            required=""
            type="date"
            value="2024-03-08"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":rb:"
          >
            Data de Criação
          </label>
        </div>
      </div>
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":rc:"
            readonly=""
            required=""
            type="date"
            value="2024-03-08"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":rc:"
          >
            Data de Atualização
          </label>
        </div>
      </div>
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            aria-label="Try Count"
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":rd:"
            name="try_count"
            required=""
            type="number"
            value="0"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":rd:"
          >
            Tentativas de Login
          </label>
        </div>
      </div>
    </div>
    <div
      class="flex w-full justify-center sm:justify-end"
    >
      <button
        aria-label="Update User"
        class="relative transition-transform duration-200 ease-in-out hover:scale-105 focus:scale-105 active:scale-90 min-w-[12.404rem]"
        type="submit"
      >
        <div
          aria-label="Button Background"
          class="absolute size-full rounded-md opacity-[16%] bg-success-500"
        />
        <div
          class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-success-500"
        >
          <span
            class="z-10 font-medium leading-[1.125rem] tracking-[0.027rem] text-[0.9375rem]"
          >
            Atualizar Colaborador
          </span>
        </div>
      </button>
    </div>
  </form>
</div>
`;

exports[`<CollaboratorGeneral /> should render the collaborator general correctly 1`] = `
<div
  class="flex flex-col gap-6 md:grid md:grid-cols-12"
>
  <div
    class="col-span-4 flex flex-col items-center justify-center gap-6 rounded-2xl border border-zumthor-100 bg-white-lilac-50 px-6 pb-10 pt-20 shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950"
  >
    <div
      class="relative flex size-36 items-center justify-center rounded-full border border-dashed border-dark-200 dark:border-shark-700"
    >
      <svg
        aria-label="Default Photo"
        class="text-comet-500 transition-all dark:text-dark-500"
        fill="none"
        height="56"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="56"
      >
        <path
          d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="12"
          cy="7"
          r="4"
        />
      </svg>
      <input
        accept="image/*"
        aria-hidden="true"
        aria-label="File"
        class="hidden"
        type="file"
      />
      <button
        aria-label="Change Photo"
        class="group absolute inset-2 z-20 flex cursor-pointer items-center justify-center rounded-full text-white-lilac-50 transition-all hover:bg-black/70 focus:bg-black/70"
        type="button"
      >
        <svg
          class="opacity-0 transition-all group-hover:opacity-100 group-focus:opacity-100"
          fill="none"
          height="36"
          viewBox="0 0 24 24"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.75992 22H17.2399C19.9999 22 21.0999 20.31 21.2299 18.25L21.7499 9.99C21.8899 7.83 20.1699 6 17.9999 6C17.3899 6 16.8299 5.65 16.5499 5.11L15.8299 3.66C15.3699 2.75 14.1699 2 13.1499 2H10.8599C9.82992 2 8.62992 2.75 8.16992 3.66L7.44992 5.11C7.16992 5.65 6.60992 6 5.99992 6C3.82992 6 2.10992 7.83 2.24992 9.99L2.76992 18.25C2.88992 20.31 3.99992 22 6.75992 22Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M10.5 8H13.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M12 18C13.79 18 15.25 16.54 15.25 14.75C15.25 12.96 13.79 11.5 12 11.5C10.21 11.5 8.75 12.96 8.75 14.75C8.75 16.54 10.21 18 12 18Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </button>
    </div>
    <span
      class="text-center text-xs text-comet-500 transition-all dark:text-dark-300"
    >
      Permitido *.jpeg, *.jpg, *.png, *.gif 
      <br />
      tamanho máximo de 5 Mb
    </span>
    <div
      class="flex gap-4"
    >
      <label
        class="relative flex cursor-pointer items-center gap-2"
        for="active"
      >
        <span
          class="text-sm text-comet-500 dark:text-gray-300"
        >
          Ativo
        </span>
        <button
          aria-checked="true"
          aria-label="Active"
          class="peer relative h-5 w-9 rounded-full bg-porcelain-100 transition-all peer-checked:after:translate-x-full peer-focus:outline-none data-[state='checked']:bg-primary-400 dark:border-gray-600 dark:bg-shark-950 data-[state='checked']:dark:bg-primary-400"
          data-state="checked"
          id="active"
          role="switch"
          type="button"
          value="on"
        >
          <span
            class="mx-[2px] flex size-4 rounded-full bg-white-lilac-50 transition-all data-[state='checked']:translate-x-full"
            data-state="checked"
          />
        </button>
      </label>
      <label
        class="relative flex cursor-pointer items-center gap-2"
        for="validated"
      >
        <span
          class="text-sm text-comet-500 dark:text-gray-300"
        >
          Validado
        </span>
        <button
          aria-checked="true"
          aria-label="Validated"
          class="peer relative h-5 w-9 rounded-full bg-porcelain-100 transition-all peer-checked:after:translate-x-full peer-focus:outline-none data-[state='checked']:bg-primary-400 dark:border-gray-600 dark:bg-shark-950 data-[state='checked']:dark:bg-primary-400"
          data-state="checked"
          id="validated"
          role="switch"
          type="button"
          value="on"
        >
          <span
            class="mx-[2px] flex size-4 rounded-full bg-white-lilac-50 transition-all data-[state='checked']:translate-x-full"
            data-state="checked"
          />
        </button>
      </label>
    </div>
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="Delete Trigger"
      class="relative transition-transform duration-200 ease-in-out hover:scale-105 focus:scale-105 active:scale-90 min-w-[9.18rem]"
      data-state="closed"
      type="button"
    >
      <div
        aria-label="Button Background"
        class="absolute size-full rounded-md opacity-[16%] bg-danger-500"
      />
      <div
        class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-danger-500"
      >
        <span
          class="z-10 font-medium leading-[1.125rem] tracking-[0.027rem] text-sm"
        >
          Deletar Usuário
        </span>
      </div>
    </button>
  </div>
  <form
    aria-label="Form"
    class="col-span-8 flex flex-col justify-between gap-6 rounded-2xl border border-zumthor-100 bg-white-lilac-50 p-6 shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950"
  >
    <div
      class="grid grid-cols-auto-fill gap-x-4 gap-y-10"
    >
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            aria-label="Document"
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":r3:"
            name="document"
            required=""
            value="123456789"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":r3:"
          >
            Documento
          </label>
        </div>
      </div>
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":r4:"
            readonly=""
            required=""
            type="date"
            value="2024-03-08"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":r4:"
          >
            Data de Criação
          </label>
        </div>
      </div>
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":r5:"
            readonly=""
            required=""
            type="date"
            value="2024-03-08"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":r5:"
          >
            Data de Atualização
          </label>
        </div>
      </div>
      <div
        class="relative flex h-min flex-col w-full"
      >
        <div
          class="relative"
        >
          <input
            aria-label="Try Count"
            autocomplete="on"
            class="peer w-full rounded-lg border-2 bg-transparent font-normal text-comet-500 outline-none transition-all placeholder:opacity-0 dark:border-shark-950 dark:text-white-lilac-50 p-3 focus:border-primary-300 dark:focus:border-primary-400 focus:placeholder:opacity-100"
            id=":r6:"
            name="try_count"
            required=""
            type="number"
            value="0"
          />
          <label
            class="pointer-events-none absolute left-4 top-1/2 z-10 -translate-y-1/2 px-[3px] text-sm text-comet-500 transition-all before:absolute before:inset-0 before:-z-10 before:block before:w-full before:bg-white-lilac-50 before:transition-all peer-valid:top-[-35%] peer-valid:translate-y-1/2 peer-read-only:top-[-35%] peer-read-only:translate-y-1/2 peer-focus:top-[-35%] peer-focus:translate-y-1/2 dark:text-dark-300 before:dark:bg-smoke-950 peer-focus:text-primary-400"
            for=":r6:"
          >
            Tentativas de Login
          </label>
        </div>
      </div>
    </div>
    <div
      class="flex w-full justify-center sm:justify-end"
    >
      <button
        aria-label="Update User"
        class="relative transition-transform duration-200 ease-in-out hover:scale-105 focus:scale-105 active:scale-90 min-w-[12.404rem]"
        type="submit"
      >
        <div
          aria-label="Button Background"
          class="absolute size-full rounded-md opacity-[16%] bg-success-500"
        />
        <div
          class="flex h-full items-center justify-center gap-2 px-5 py-[0.625rem] flex-row text-success-500"
        >
          <span
            class="z-10 font-medium leading-[1.125rem] tracking-[0.027rem] text-[0.9375rem]"
          >
            Atualizar Colaborador
          </span>
        </div>
      </button>
    </div>
  </form>
</div>
`;
